File: solutions/1441. Build an Array With Stack Operations/1441.js
// Time complexity: O(mn)
// Space complexity: O(m)
var buildArray = function(target, n) {
    let ans = [];
    let i = 0;

    for (let num of target) {
      while (i < num - 1) {
        ans.push("Push");
        ans.push("Pop");
        i++;
      }

      ans.push("Push");
      i++;
    }

    return ans;
};

File: solutions/1441. Build an Array With Stack Operations/1441.cs
// Time complexity: O(mn)
// Space complexity: O(m)
public class Solution {
    public IList<string> BuildArray(int[] target, int n) {
        List<string> ans = new List<string>();
        int i = 0;

        foreach (int num in target)
        {
            while (i < num - 1)
            {
                ans.Add("Push");
                ans.Add("Pop");
                i++;
            }

            ans.Add("Push");
            i++;
        }

        return ans;
    }
}